compiler/src/dotty/tools/dotc/config/ScalaSettings.scala

```diff
@@ -158,43 +158,54 @@ private sealed trait WarningSettings:
 
   val Whelp: Setting[Boolean] = BooleanSetting(WarningSetting, "W", "Print a synopsis of warning options.")
   val XfatalWarnings: Setting[Boolean] = BooleanSetting(WarningSetting, "Werror", "Fail the compilation if there are any warnings.", aliases = List("-Xfatal-warnings"))
-  val WvalueDiscard: Setting[Boolean] = BooleanSetting(WarningSetting, "Wvalue-discard", "Warn when non-Unit expression results are unused.")
-  val WNonUnitStatement = BooleanSetting(WarningSetting, "Wnonunit-statement", "Warn when block statements are non-Unit expressions.")
-  val WenumCommentDiscard = BooleanSetting(WarningSetting, "Wenum-comment-discard", "Warn when a comment ambiguously assigned to multiple enum cases is discarded.")
-  val WimplausiblePatterns = BooleanSetting(WarningSetting, "Wimplausible-patterns", "Warn if comparison with a pattern value looks like it might always fail.")
-  val WunstableInlineAccessors = BooleanSetting(WarningSetting, "WunstableInlineAccessors", "Warn an inline methods has references to non-stable binary APIs.")
+  val Wall: Setting[Boolean] = BooleanSetting(WarningSetting, "Wall", "Enable all warnings.")
+  private val overrideWithWall: (Setting[Boolean], (Boolean, Any) => Boolean) =
```

Comment:
I feel `overrideWithWall` and all the warning settings are over-engineered. I think a better solution is to create an object which reads all warning settings properly and stores in ctx.

Reply:
thanks for the suggestion! I'd appreciate some clarification: does the proposed solution mean that we create another object that takes `Wall` into consideration when reading all warning settings and store it in the context, (similar to `WunusedHas`)? Then at all use sites of these warning settings we read from that object instead?